main.go

```diff
@@ -388,12 +388,12 @@ func setWithEnvVariables(options *config.Config) {
 func setupApp(config *config.Config) (a *glusterfs.App) {
 	defer func() {
 		err := recover()
-		if a == nil {
-			fmt.Fprintln(os.Stderr, "ERROR: Unable to start application")
-			os.Exit(1)
-		} else if err != nil {
+		if err != nil {
 			fmt.Fprintf(os.Stderr, "ERROR: Unable to start application: %s\n", err)
 			os.Exit(1)
+		} else if a == nil {
+			fmt.Fprintln(os.Stderr, "ERROR: Unable to start application")
```

Comment:
Shouldn't this case be unreachable with the refactoring?

Reply:
yes. Will update